### ISSUE TITLE
Evaluate postcss or mensch as an alternative CSS parser

### ISSUE THREAD

**Issue by papandreou** [assetgraph/assetgraph#195]
https://github.com/brettstimmerman/mensch

Seems like it'd allow us to get rid of the multiple-definitions-of-the-same-property hack, preserve comments, and allow us to report line/column numbers of CSS rules/selectors/properties/etc.

**Comment by papandreou** [comment]
I think postcss is where it's at, especially if we want source map support.

**Comment by papandreou** [comment]
Figured out how to use postcss' parser and get a separate source map out when serializing:

```js
global.Promise = require('rsvp').Promise;

var postcss = require('postcss');

var parseTree = postcss.parse('body {color: red; background-color: green}', { source: 'somewhereOverTheRainbow.css' });
console.log(parseTree.toResult({ map: { inline: false } }));
```

**Comment by papandreou** [comment]
The `map` options object can also supports an `annotation` property that governs whether the `sourceMappingURL` comment gets added. `false` disables it, a string value specifies its url:

```js
parseTree.toResult({ map: { inline: false, annotation: 'whatever.map' } });
```

**Comment by Munter** [comment]
I guess the only downside is that we lose the CSSOM model. On the other hand I don't know any that actually ever used it, so it's probably ok. And we'll get a whole bunch of extra post process scripts for absolutely free by existing postcss authors!

**Comment by papandreou** [comment]
Yeah, exactly. All code that manipulates CSS via `cssAsset.parseTree` will be AssetGraph-specific anyway. Jsdom also provides `htmlElement.style` and `document.styleSheets` with a CSSOM-interface. I guess we could try to interop with that (and reflect changes in the underlying asset), but that's a separate issue anyway.

**Comment by papandreou** [comment]
Addressed in #326
